svelte/src/scripts/draw/helper/custom-d3-forces.ts

```diff
@@ -1,166 +1,222 @@
-						const angle = Math.atan(delta);
-						const vy = Math.sin(angle);
-						const vx = Math.cos(angle);
-
-						// Set the values
-						if (node1.middle.x > node2.middle.x) {
-							nodes[i].vx! += vx * n1ForceStrength;
-							nodes[i].vy! += vy * n1ForceStrength;
-							nodes[j].vx! -= vx * n2ForceStrength;
-							nodes[j].vy! -= vy * n2ForceStrength;
-						} else {
-							nodes[i].vx! -= vx * n1ForceStrength;
-							nodes[i].vy! -= vy * n1ForceStrength;
-							nodes[j].vx! += vx * n2ForceStrength;
-							nodes[j].vy! += vy * n2ForceStrength;
-						}
-					}
-				}
+                if (nodes[i].x && nodes[i].y && nodes[j].x && nodes[j].y) {
+                    // Caclulate x and y position of the top-left and bottom-right corner of the node, and middle position.
+                    const node1 = makeSquare(nodes[i]);
+				    const node2 = makeSquare(nodes[j]);
+                    
+                    // Check for collision
+                    if (node1.bottomright.x >= node2.topleft.x && node1.topleft.x <= node2.bottomright.x && 
+                        node1.bottomright.y >= node2.topleft.y && node1.topleft.y <= node2.bottomright.y) {
+                        // We collided!
+                        if (snap) {
+                            // Snap stuff into place
+                            if (node2.middle.x < node1.middle.x) {
+                                nodes[i].x = nodes[j].x! + 0.5 * nodes[j].width + 0.5 * nodes[i].width;
+                            } else {
+                                nodes[j].x = nodes[i].x! + 0.5 * nodes[i].width + 0.5 * nodes[j].width;
+                            }
+
+                            if (node2.middle.y < node1.middle.y) {
+                                nodes[i].y = nodes[j].y! + 0.5 * nodes[j].height + 0.5 * nodes[i].height;
+                            } else {
+                                nodes[j].y = nodes[i].y! + 0.5 * nodes[i].height + 0.5 * nodes[j].height;
+                            }
+                        }
+                        else {
+                            // Use a force instead
+                            chargeForce(node1, node2, alpha, i, j, 20);
+                        }
+                    } else if (
+                        // We also want to try to keep a minimum distance. However, for performance reasons we don't do this on the innermost level
+                        nodes[i].level === nodes[j].level && nodes[i].members!.length > 0 && nodes[j].members!.length > 0
+                    ) {
+                        const clearDistance = Math.max(
+                            Math.abs(node1.middle.x - node2.middle.x) - ((node1.width + node2.width)/2),
+                            Math.abs(node1.middle.y - node2.middle.y) - ((node1.height + node2.height)/2),
+                        );
+
+                        chargeForce(node1, node2, alpha, i, j, 300*(1/clearDistance));
```

Comment:
hmm not sure again how this one is gonna looks like. I'll check it on the browser after merging